codec-http/src/main/java/io/netty/handler/codec/http/HttpContentCompressor.java

```diff
@@ -199,6 +203,8 @@ public HttpContentCompressor(int contentSizeThreshold, CompressionOptions... com
         this.gzipOptions = gzipOptions;
         this.deflateOptions = deflateOptions;
         this.zstdOptions = zstdOptions;
+        this.factories = new CompressionEncoderFactory[]{new GzipEncoderFactory(), new DeflateEncoderFactory(),
```

Comment:
I think you could share the same static instance as it is stateless

Reply:
Like the reason before, the map here cannot be static